File: app/src/main/java/com/example/administrator/baseproject/MainActivity.java
package com.example.administrator.baseproject;

import android.os.Bundle;
import android.view.View;

import com.example.administrator.baseproject.bean.TestBean;
import com.example.administrator.baseproject.databinding.ActivityMainBinding;
import com.htjy.baselibrary.base.BaseAcitvity;
import com.htjy.baselibrary.bean.JavaBaseBean;
import com.htjy.baselibrary.http.base.JsonDialogCallback;
import com.htjy.baselibrary.utils.DialogUtils;
import com.htjy.baselibrary.utils.LogUtils;
import com.lzy.okgo.OkGo;
import com.lzy.okgo.model.Response;

public class MainActivity extends BaseAcitvity {

    private ActivityMainBinding mainBinding;

    @Override
    protected boolean isBinding() {
        return true;
    }

    @Override
    protected void setContentViewByBinding(int layoutId) {
        mainBinding = getContentViewByBinding(layoutId);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initViews(Bundle savedInstanceState) {

        mainBinding.tabLayout.addTab(mainBinding.tabLayout.newTab().setText("test1"));
        mainBinding.tabLayout.addTab(mainBinding.tabLayout.newTab().setText("test2"));
        mainBinding.viewpager.setAdapter(new TestFragmentAdapter(getSupportFragmentManager(), "test1", "test2"));
        mainBinding.tabLayout.setupWithViewPager(mainBinding.viewpager);

        mainBinding.tvTest1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DialogUtils.showSimpleDialog(activity, "000", null);
                DialogUtils.showConfirmDialog(activity, "123", "456", null, null);
            }
        });
        mainBinding.tvTest2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                OkGo.<JavaBaseBean<TestBean>>get("http://www.baokaodaxue.com/yd/Bkdxsecond_Startload/getConfig")
                        .tag(this)
                        .params("app", "1")
                        .params("type", "1")
                        .execute(new JsonDialogCallback<JavaBaseBean<TestBean>>(MainActivity.this) {
                            @Override
                            public void onSimpleSuccess(Response<JavaBaseBean<TestBean>> response) {
                                super.onSimpleSuccess(response);
                                LogUtils.d(response.body().getMessage().toString());
                            }
                        });
            }
        });
    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initListener() {

    }

    @Override
    protected void initStateLayout() {

    }

    @Override
    public void showNullLayout() {

    }

    @Override
    public void showErrorLayout() {

    }

    @Override
    public void showSuccessLayout() {

    }

    @Override
    public void showNetWorkErrorLayout() {

    }
}

File: app/src/main/java/com/example/administrator/baseproject/MainActivity.kt
package com.example.administrator.baseproject

import android.os.Bundle
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.viewpager.widget.ViewPager
import com.example.administrator.baseproject.databinding.ActivityMainBinding
import com.google.android.material.tabs.TabLayout
import com.htjy.baselibrary.base.ViewModelActivity

class MainActivity : ViewModelActivity<ActivityMainBinding,MainViewModel>() {


    /*override fun setContentViewByBinding(layoutId: Int) {
        binding = getContentViewByBinding(layoutId)
    }*/


    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }


    override fun initViews(savedInstanceState: Bundle?) {
        val iv = findViewById<ImageView>(R.id.iv_head)
        val tv_test1 = findViewById<TextView>(R.id.tv_test1)
        val tv_test2 = findViewById<TextView>(R.id.tv_test2)
        val frameLayout = findViewById<FrameLayout>(R.id.fl_content)
        val viewpager = findViewById<ViewPager>(R.id.viewpager)
        val mTabLayout = findViewById<View>(R.id.tab_layout) as TabLayout
        mTabLayout.addTab(mTabLayout.newTab().setText("test1"))
        mTabLayout.addTab(mTabLayout.newTab().setText("test2"))
        viewpager.adapter = TestFragmentAdapter(supportFragmentManager, "test1", "test2")
        mTabLayout.setupWithViewPager(viewpager)


        mBinding.viewModel = mViewModel
        /*showFragment(Test1Fragment.class);
        tv_test1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showFragment(Test1Fragment.class);
            }
        });

        tv_test2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showFragment(Test2Fragment.class);
            }
        });*/
        //ImageLoaderUtil.getInstance().loadCircleBorderImage("http://img.taopic.com/uploads/allimg/140729/240450-140HZP45790.jpg",R.color.transparent
        //,iv,20,getResources().getColor(R.color.red));
        tv_test1.setOnClickListener {
            // ImageLoaderUtil.getInstance().loadCircleBorderImage("http://img.taopic.com/uploads/allimg/140729/240450-140HZP45790.jpg",R.color.transparent
            //          ,iv,20,getResources().getColor(R.color.red));
        }
    }
    override fun initData() {
        mViewModel.getData()
        mBinding.tvTest.postDelayed({
            mViewModel.updateData()
        }, 1000)

    }
    override fun initListener() {}
}